File: solutions/clojure/grains/1/src/grains.clj
(ns grains)

(defn square [cell]
  ;; (bit-shift-left 1 (dec 1)) ;; this number is a big boy
  (.shiftLeft BigInteger/ONE (dec cell)))

;; (map square (range 1 8))

;; (square 64)

(defn total []
  (dec (square 65)))

File: solutions/go/grains/1/grains.go
package grains

import "errors"

var errInvalidInput = errors.New("number must be in range [1, 64]")

func Square(number int) (uint64, error) {
	if number < 1 || number > 64 {
        return 0, errInvalidInput
    }
    return 1 << (number - 1), nil
}

func Total() (total uint64) {
	for i := 1; i <= 64; i++ {
        sq, _ := Square(i)
        total |= sq
    }
    return
}
